collectors/jiraffe/convertors.py

```diff
@@ -16,6 +17,50 @@
 )
 
 
+class TrackerSaver:
+    """
+    TrackerSaver is holder of the individual tracker parts provided by TrackerConvertor
+    which knows how to correctly save and link them all as the resulting Django DB models
+    it provides save method as an interface to perform the whole save operation
+    """
+
+    def __init__(
+        self,
+        tracker,
+        affects,
+        alerts,
+    ):
+        self.tracker = tracker
+        self.affects = affects
+        self.alerts = alerts
+
+    def __str__(self):
+        return f"TrakcerSaver {self.tracker.type}:{self.tracker.external_system_id}"
+
+    def save(self):
+        """
+        save the tracker with its context to DB
+        """
+        # wrap this in an atomic transaction so that
+        # we don't query this tracker during the process
+        with transaction.atomic():
+            # re-create all affect links
+            # in case some where removed
+            self.tracker.affects.clear()
+            self.tracker.affects.add(*self.affects)  # bulk add
+            self.tracker.save(
```

Comment:
Suggesting to add a comment:
```
# NOTE: When enabling independent read & write DB replicas, this might break, because the just-written values will be read before they can be propagated. This will again cause something like OSIDB-2487.
```

Reply:
I am not sure that it is the issue of this code. I would consider it as an issue of the replica logic/setup. The atomic transaction is declared and I do not think that Python/Django developer should care here for how it is done in the database/Django framework. Without the atomicity someone else might get inconsistent data.

Reply:
@JimFuller-RedHat FYI
@osoukup :+1: